llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2271,6 +2271,27 @@ Instruction *InstCombinerImpl::foldVectorBinop(BinaryOperator &Inst) {
     }
   }
 
+  // Similar to the combine above, but handles the case for scalable vectors
+  // where both V1 and C are splats.
```

Comment:
```suggestion
  // where both shuffle(V1, 0) and C are splats.
```
`V1` is not required to be a splat vector.